src/User/Command/RegisterUserHandler.php

```diff
@@ -134,8 +136,25 @@ private function applyToken(User $user, RegistrationToken $token)
         );
     }
 
+    /**
+     * @throws InvalidArgumentException
+     */
     private function uploadAvatarFromUrl(User $user, string $url)
     {
+        $urlValidator = Validator::make(compact('url'), [
+            'url' => 'required|active_url',
+        ]);
+
+        if ($urlValidator->fails()) {
+            throw new InvalidArgumentException('Provided avatar URL must be a valid URI.', 503);
+        };
+
+        $parsed_url = parse_url($url);
+
+        if (!($parsed_url['scheme'] === 'http' || $parsed_url['scheme'] === 'https')) {
+            throw new InvalidArgumentException('Provided avatar URL must have scheme http or https. Scheme provided was "' . $parsed_url['scheme'] . '".', 503);
+        }
```

Comment:
What about this to make things slightly simpler?
```suggestion
        $scheme = parse_url($url, PHP_URL_SCHEME);

        if (! in_array($scheme, ['http', 'https'])) {
            throw new InvalidArgumentException("Provided avatar URL must have scheme http or https. Scheme provided was $scheme.", 503);
        }
```
Also I think this needs to throw a ValidationException with an array `[$attribute=>$reason]` so that the frontend can properly render it? @datitisev or @SychO9 please confirm?

Reply:
We discussed that but thought that displaying that would make users confused as it would seem as if they've done something wrong, when they have no control over that input.

Reply:
Cool makes sense!